docs/blog/2022-07-20-bun-is-the-future.mdx

```diff
@@ -0,0 +1,239 @@
+the [io.js fiasco](https://thenextweb.com/news/Node-js-and-io-js-are-settling-their-differences-merging-back-together),
+know that a fork with a simple increase in interpreter speed has been enough to shake the entire ecosystem in the past.
+Bun also starts much faster than Node. Around 7ms for me, which makes it around 10x faster than starting Node. This makes it a great fit for serverless environments
+and things at the edge.
+
+This time, speed isn't the only difference. Bun also adds a lot of standard-library functions that
+are nice to have. `Bun.write()`, for instance, is a new function for writing files that returns a promise and claims to be faster
+by using more appropriate system calls. No more calling `util.promisify(fs.writefile)` just to get a function you can `await`.
+It also finally has `fetch` and `websocket` built in. Nice.
+
+While we're on the topic of Node APIs, Bun currently supports around 90% of Node's existing APIs. Node is huge and can do some
+things almost nobody knows about [(`new AsyncLocalStorage()` anyone?)](https://github.com/lunasec-io/lunasec/blob/813d0abe1587a86bd7f89eea2b9e847f208b65d3/lunatrace/bsl/logger/src/index.ts#L80), so that's pretty impressive. Can you run *every*
+package you find on NPM? No, but it does seem to work most of the time.
+
+
+
+### TypeScript
+By the way, TypeScript is a first class citizen with Bun. Simply call `bun my-ts-file.ts`. It just works. This is similar to Deno's support for TS.
+Template out a new project with Bun or add `bun-types` to your tsconfig and your IDE autocomplete will work fine for those new functions!
+
+I'm not sure if Bun is also a complete rewrite of the TypeScript compiler, but it appears to be. At this point I'm not surprised. Some more advanced TypeScript
+configuration and features are still unsupported, such as decorators and the `extends` feature in tsconfig for merging multiple configs together.
+
+## The package manager
+
+Among the most exciting of its talents, Bun is a drop-in replacement for NPM. It's really, really fast, and that's about what I have to say about it.
+It works how you would expect.
+
+> On Linux, bun install tends to install packages 20x - 100x faster than npm install. On macOS, it’s more like 4x - 80x.
+
+I can confirm, it's faster with no cache, it's faster with a full cache, it's just faster.
+
+This isn't the first time someone has tried to speed up NPM. You might be familiar with Yarn's Plug-n-Play and how it completely ditches the `node_modules` folder to try to get
+faster installs.  It does *work* but in practice,
+it's really hard to get working, and requires a lot of polyfilling and escape-hatches. We bit-the-bullet to use it, but I probably wouldn't recommend it or use it again.
+
+Bun works the way NPM does, just faster. It uses its own lockfile format, but node_modules and the package.json look normal.
```

Comment:
not required but it would be cool to talk about _why_ the bun package manager is faster while working on the same dataset